src/logging/log.cpp

```diff
@@ -71,3 +71,8 @@ void mpl::set_logger(std::shared_ptr<Logger> logger)
     global_logger = std::move(logger);
     qInstallMessageHandler(qt_message_handler);
 }
+
+auto mpl::get_logger() -> Logger* // for tests, don't rely on it lasting
```

Comment:
I'm curious why you use the `auto blah -> Logger*` format here instead of just a straight up `Logger* blah` like in the function definition?  I know they are treated one and the same and I'm not going to block this at all, but I'm just wondering if there is some sort of optimization I don't know about :smile:

Reply:
It's just that the trailing return type is automatically looked up in the function's scope (like arguments). So you can say `auto ns::f() -> foo` whereas the namespace is needed if `foo` comes first: `ns::foo ns::f()`. It feels like a little redundancy that can be economized, so I usually do it like that. When the ns is small it may be more typing because of the added auto.

Reply:
Ah, cool, yeah, that certainly makes it easier to code for without sacrificing readability.